\begin{abstract}
\section{Motivation:}
Genome-wide association studies can potentially unravel the mechanisms behind complex traits and common genetic diseases. Despite the valuable results produced thus far, many questions remain unanswered. For instance, which specific genetic compounds are linked to the risk of the disease under investigation; what biological mechanism do they act through; or how do they interact with environmental and other external factors? 
The driving force of computational biology is the constantly growing amount of big data generated by high-throughput technologies. A practical framework that can deal with this abundance of information and that consent to discovering genetic associations and interactions is provided by means of networks. 
Unfortunately, high dimensionality, the presence of noise and the geometry of data can make the aforementioned problem extremely challenging.
We propose a penalised linear regression approach that can deal with the aforementioned issues that affect genetic data. We analyse the gene expression profiles of individuals with a common trait to infer the network structure of interactions among genes. 
\section{Results:}
The permutation-based approach leads to more stable and reliable networks inferred from synthetic microarray data. We show that a higher number of permutations determines the number of predicted edges, improves the overall sensitivity and controls the number of false positives. 

\section{Availability:}
availability of source code upon request

\section{Contact:} \href{francesco.gadaleta@ulg.ac.be}{francesco.gadaleta@ulg.ac.be}
\end{abstract}
