\section{Introduction}
Any biological system is characterised by interactions between components. The study of these interactions is essential to understanding the mechanisms that regulate complex diseases and to unravel the functional aspects of genetic compounds. 
In several fields of research, from social to telecommunication and biology, system interactions are more and more often represented by graphical models (\citealp{Vidal2011Complex, BAR03a, wisdomcrowds}). Generally speaking, those are defined by a set of nodes and a set of edges. Each node usually represents a specific biological component that interacts with others to perform specific functions. Edges may have several meanings, depending on the type of interactions it represents, such as similarity, causality, distance, etc. 
In the field of network theory and genetics, the nodes of a graph usually represent genes and the edges represent the interactions among these genes. Consequently, a network graph of genetic interactions is a viable way to visualise clusters, modules or pathways, according to the purpose of the analysis.
It is known that genes act in clusters and their individual effects tend to be characterised by a smaller magnitude within the system as a whole (Refs.) Graphical models facilitate the detection of the main genetic effects. Moreover, pathways of genes become more visible to the researcher who investigates the data, giving a more complete explanation of the biological function that the pathway itself performs.
One viable way to represent the interactions of the nodes of a graph - and consequently the  topology of the resulting network - might be represented by the adjacency matrix $\beta = \beta_{ij}$. The values of each cell $(i,j)$ in the adjacency matrix represent the magnitude of the interaction between two nodes, whereas zeros are equivalent to absence of interaction between node $i$ and node $j$. 
Specifically to the field of computational biology, the problem of learning the structure of a graph starting from the expression profile of a number of genes is a challenging task due to several reasons that are already known to the research community. The presence of noise in the measurements, the high dimensionality of genetic data and multicollinearity are just a few that are mentioned.
 
Despite active research in the field of high-density oligonucleotide arrays, noise still represents a consistent source of error. Any analysis subsequent to the measurement of a subset of genes should take into consideration the artifacts introduced by noise or by the computational method that mitigates it (\citealp{microarray_noise, microarray_high_noise}).
   
It is very common to perform analyses over high-dimensional genetic data in which the number of genes $p$ is much larger than the number of individuals $n$. Discovering the interactions between genes in such cases is extremely difficult. Methods that rely on the dispersion matrix whose elements $(i,j)$ represent the covariance between the expressions of two genes are also affected by high-dimensionality. As a matter of fact, the maximum likelihood estimator cannot provide a reliable estimate of the covariance matrix for $n<<p$ problems. 

Moreover, gene expression profiles are affected by the presence of multicollinearity (\citealp{est_multicoll, ml_multicoll}), namely two or more genes or genetic compounds can be highly correlated. The presence of multicollinearity can influence the performance of regression-based models. The regression coefficient of a predictor variable's importance on the target variable has the tendency to lose precision with respect to the case in which the same genes were uncorrelated. 
From a biological perspective, it is broadly recognised that strong genetic correlations are utterly frequent in microarray data and that, in contrast, complete independence between any two gene expression measurements is rare (refs). Therefore, it is expected that functionally related genes are somehow co-expressed. This phenomenon can be expressed by assuming the presence of high correlation for a subset of genes in the dataset under study. Moreover, as the gene sets to be tested are usually chosen on the basis of functional annotation, it should be expected that many of the tested genes might be, in fact, correlated (\citealp{genesets}).
Variable selection methods are even more sensitive to the presence of multicollinearity as they tend to select only one or few highly correlated variables.

A computational approach that can deal with the aforementioned issues affecting genetic interactions belongs to the family of penalised linear regression. The core idea consists in reducing the number of meaningful interactions with each gene, in order to build sparser networks. Penalised linear regression has been investigated in seminal work reported in (\citealp{Tibshirani94regressionshrinkage, Meinshausen06highdimensional, finegold, Meinshausen_stabilityselection}), in which each variable is considered response and the remaining ones are independent covariates. In the aforementioned work, bootstrapping has been extensively used to improve the stability of the predicted interactions. Unfortunately, the nature of genetic data and the presence of highly correlated variables play a detrimental role that can affect the overall reliability of these methods. Moreover, Lasso-based regression procedures are known to deal poorly with highly correlated variables since only one in a group of multi correlated covariates is selected. Bootstrapping does not seem to mitigate such a troublesome condition.

In this paper, we consider the use of Lasso penalised regression as a starting point. We subsequently rely on a permutation-based approach in order to increase the significance of the discovered interactions. 

In Section \ref{approach}, we describe the method in detail. In Section \ref{results}, we measure the performance of our approach on simulated genetic networks of different size. Conclusion and future developments are drawn in Section \ref{conclusion}.

%However, it remains unclear how the three processes of differentiation, proliferation, and apoptosis in regulating stem cells collectively manage these challenging tasks.



