\documentclass[11pt, oneside]{article}   	% use "amsart" instead of "article" for AMSLaTeX format
\usepackage{geometry}                		% See geometry.pdf to learn the layout options. There are lots.
\geometry{letterpaper}                   		% ... or a4paper or a5paper or ... 
%\geometry{landscape}                		% Activate for for rotated page geometry
%\usepackage[parfill]{parskip}    		% Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}				% Use pdf, png, jpg, or epsÂ§ with pdflatex; use eps in DVI mode
								% TeX will automatically convert eps --> pdf in pdflatex		
\usepackage{amssymb}
\usepackage{authblk}


\title{A network-based omics integration framework:\\
 overcoming the obstacle of high dimensional data}

        
\author[1,2]{Francesco Gadaleta}
\author[1,2]{Kyrylo Bessonov}
\author[1,2]{Kridsadakorn Chaichoompu}
\author[3]{Silvia Pineda}
\author[1,2]{Kristel Van Steen}


\affil[1]{Systems and Modeling Unit, Montefiore Institute, University of Liege, Belgium}
\affil[2]{Bioinformatics and Modeling, GIGA-R, University of Liege, Belgium}
\affil[3]{Centro Nacional de Investigaciones Oncologicas, Spain}

\affil[1]{\textit {\{francesco.gadaleta, kristel.vansteen\}@ulg.ac.be}}


\date{}

\begin{document}
\maketitle

\begin{abstract}
Genome-wide association studies can potentially unravel the mechanisms behind complex traits and common genetic diseases. Despite the valuable results produced thus far, many questions remain unanswered. For instance, which specific common variants are linked to the risk of
the disease under investigation, what biological mechanism do they act through or how do they interact with environmental and other external
factors? 

The driving force of computational biology is the constantly growing amount of big data generated by high-throughput technologies. The amount of available data and its heterogeneity seem to play a beneficial role rather than a detrimental one in discovering new genetic insights. Each type of data, e.g. gene expression, epigenetic, methylation, RNAseq or proteomics, provided by a diverse source of information, directly contributes to complete the overall puzzling picture of genetic disorders, with its own unique local knowledge. In such a scenario, data integration, as the practice of combining evidence from different data sources, represents the most challenging activity, due to the unattainable task of merging large and heterogeneous data sets.

A practical framework that fulfils the needs of integration is provided by means of networks. Due to the manifest risks of introducing bias when integrating heterogeneous data and because of the curse of dimensionality, which is a common aspect in computational biology, preliminary procedures of variable selection are essential. 
%We investigate two approaches that capture the multidimensional relationships between different data sets, namely conditional inference trees and linear regression methods. 
A preliminary research strategy consists in using every expression trait as a dependent variable and the remaining expression traits as covariates. The aforementioned strategy will derive variable importance scores that, in turn lead to selecting important variables. We subsequently proceed with the construction of weighted networks that integrate evidence gathered from different data sets with the purpose of detecting pathways and interactions among genetic compounds and better describing the complex mechanisms behind the traits under study.
Results on synthetic data show that the described method not only suggests the main covariates associated with the response, but can also detect the network components that are potentially implicated in the disease under study.


% should speak about results here

%The diversity of data
%The diversity of methods 
%From data to networks

%These big data is a gold mine for new discoveries, new patterns, and new hypotheses. At the same time, the data is heterogeneous with low signal-to-noise ratio. Handling such big data requires advanced, efficient and accurate computational methods and computer tools.

\end{abstract}

\smallskip
\noindent \textbf{Keywords.} omics integration, network, regression


%\section{}
%\subsection{}

%\section{Introduction}

%\paragraph{Outline}
%The remainder of this article is organised as follows.
%Section~\ref{previous work} gives account of related work. 
%Our new and exciting results are described in Section~\ref{results}.
%Finally, Section~\ref{conclusions} gives the conclusions.

%\section{Related work}\label{related word}
%TODO

%\section{Results}\label{results}
%TODO 

%\section{Conclusions}\label{conclusions}
%TODO

%\bibliographystyle{abbrv}
%\bibliography{simple}

\end{document}

This is never printed
